components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementPolicyAdminService.java

```diff
@@ -437,59 +428,64 @@ public PublisherDataHolder getSubscriber(String subscribeId) throws EntitlementE
      * @throws EntitlementException throws, if fails
      */
     public String[] getSubscriberIds(String searchString) throws EntitlementException {
-        PolicyPublisher publisher = EntitlementAdminEngine.getInstance().getPolicyPublisher();
```

Comment:
Should handle similar to https://github.com/wso2/carbon-identity-framework/pull/5673/files#r1594986499

Reply:
check the rest as well